### PR TITLE
Adding command line exclusion lists, so that users can prune fields from output. 

### LINKED ISSUE #4359: Funcotator - Need to have a funcotation output list
Must allow support for an output funcotation list that, if specified, will cause Funcotator only output funcotations from the given list and suppress all other funcotations.

### LINKED ISSUE #4693: Funcotator: deal with delimiters in data for output formats
Eg., need to escape or otherwise deal with `;`, `,`, `|`, etc. characters in datasource values when writing to VCF.

### PR DESCRIPTION
**Pull request by LeeTL1220**
- Added support for `--exclusion-list` parameter, which can be specified multiple times.  This will remove some rendered fields from the output.  The field should appear exactly as written in the output MAF or VCF (e.g. ClinVar_ALLELEID). Closes #4359. 
- Does basic cleaning of field values for MAF.  I.e. cleaning tabs and \n.  Fixes #4693

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorArgumentDefinitions.java [hunk 1]
```diff
@@ -48,6 +48,11 @@
      */
     public static final String ANNOTATION_OVERRIDES_LONG_NAME = "annotation-override";
 
+    /**
+     * List of final rendered fields to exclude from a final rendering
+     */
+    public static final String EXCLUSION_FIELDS_LONG_NAME = "exclusion-list";
```
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorArgumentDefinitions.java]: I'm not sure `exclusion-list` is a clear name.  Maybe something like "--skip-field" would be clearer.

File: src/test/java/org/broadinstitute/hellbender/utils/test/FuncotatorTestUtils.java [hunk 2]
```diff
@@ -49,4 +59,114 @@ public static FeatureContext createFeatureContext(final List<DataSourceFuncotati
         return FeatureContext.createFeatureContextForTesting(featureInputsWithType, dummyToolInstanceName, interval,
                 featureQueryLookahead, cloudPrefetchBuffer, cloudIndexPrefetchBuffer, reference);
     }
+
+    /**
+     * Create a dummy gencode funcotation in the most manual way possible.  This method does not check for consistency
+     *  or reasonable values.
+     *
+     * @param hugoSymbol
```
Comment (jonn-smith) [inline src/test/java/org/broadinstitute/hellbender/utils/test/FuncotatorTestUtils.java]: It's kind of a pain, but can you fill in some descriptions for these fields?

**Comment by jonn-smith** [comment]
Back to you, @LeeTL1220

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorArgumentDefinitions.java (hunk 1 above)
Comment (LeeTL1220) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorArgumentDefinitions.java]: I like `--exclude-field`

Done

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java [hunk 3]
```diff
@@ -2175,5 +2186,24 @@ public static boolean areAnyGencodeFuncotation(final List<Funcotation> funcotati
...
     }
+
+    /**
+     * @param funcotation Funcotation to render for a VCF.  Never {@code null}
+     * @param includedFields List of fields to include.  Any that match fields in the funcotation will be rendered.
+     *                       Never {@code null}
+     * @return string with the VCF representation of a {@link VcfOutputRenderer#FUNCOTATOR_VCF_FIELD_NAME} for the given
+     * Funcotation.  Never {@code null}, but empty string is possible.
+     */
+    public static String renderSanitizedFuncotationForVcf(final Funcotation funcotation, final List<String> includedFields) {
```
[the remark this answers is not in the document]
Comment (LeeTL1220) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java]: So this is an excellent question.  I originally had `renderSanitizedFuncotationForMaf`, but realized that I did not need it.

For VCFs, we render each funcotation separately, then concatenate strings.  This approach has the drawback of being less flexible in terms of ordering the fields (and setting up aliases), but that has not mattered yet.  Regardless, it means that all string operations (e.g. excluding fields and sanitizing values) must be in the same method (in this case `renderSanitizedFuncotationForVcf`) and that method must work on a funcotation.

For MAFs, we flatten out the funcotations and put the fields into a giant map.  Then we do the changes to field names and values on that map.  But by the time I want to exclude fields and sanitize, the map is already made, so we do not render individual funcotations.  Therefore no need for a `renderSantiziedFuncotationForMaf`.

If you would like to refactor to make the conventions similar, we can file an issue.  

No action.

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/mafOutput/MafOutputRenderer.java [hunk 4]
```diff
@@ -254,7 +259,7 @@ public MafOutputRenderer(final Path outputFilePath,
         defaultMap.put(MafOutputRendererConstants.FieldName_BAM_File, MafOutputRendererConstants.UNUSED_STRING);
 
         // Cache the manual annotation string so we can pass it easily into any Funcotations:
-        manualAnnotationSerializedString = (manualAnnotations.size() != 0 ? MafOutputRendererConstants.FIELD_DELIMITER + String.join( MafOutputRendererConstants.FIELD_DELIMITER, manualAnnotations.values() ) + MafOutputRendererConstants.FIELD_DELIMITER : "");
+        //manualAnnotationSerializedString = (manualAnnotations.size() != 0 ? MafOutputRendererConstants.FIELD_DELIMITER + String.join( MafOutputRendererConstants.FIELD_DELIMITER, manualAnnotations.values() ) + MafOutputRendererConstants.FIELD_DELIMITER : "");
```
[the remark this answers is not in the document]
Comment (LeeTL1220) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/mafOutput/MafOutputRenderer.java]: Done

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/mafOutput/MafOutputRenderer.java [hunk 5]
```diff
@@ -327,12 +334,14 @@ public void write(final VariantContext variant, final FuncotationMap txToFuncota
                     writeString(entry.getValue());
                     writeString(MafOutputRendererConstants.FIELD_DELIMITER);
                 }
-                writeLine(manualAnnotationSerializedString);
+//                writeLine(manualAnnotationSerializedString);
```
[the remark this answers is not in the document]
Comment (LeeTL1220) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/mafOutput/MafOutputRenderer.java]: Done

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorArgumentCollection.java [hunk 6]
```diff
@@ -101,4 +101,11 @@
             doc = "(Advanced / DO NOT USE*) If you select this flag, Funcotator will force a conversion of variant contig names from b37 to hg19.  *This option is useful in integration tests (written by devs) only."
     )
     public boolean forceB37ToHg19ContigNameConversion = false;
+
+    @Argument(
```
[the remark this answers is not in the document]
Comment (LeeTL1220) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorArgumentCollection.java]: Done

File: src/test/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtilsUnitTest.java [hunk 7]
```diff
@@ -1610,4 +1611,55 @@ public void testCreateFuncotationsFromVariantContext(final VariantContext vc, fi
...
+        Assert.assertEquals(guess, gt);
+
+    }
+
+    @SuppressWarnings("unchecked")
+    @DataProvider
+    public Object[][] provideForRenderSanitizedFuncotationForVcf() {
+
+        return new Object[][]{
+                {OutputRenderer.createFuncotationFromLinkedHashMap(
```
[the remark this answers is not in the document]
Comment (LeeTL1220) [reply src/test/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtilsUnitTest.java]: Good idea.  Done

File: src/test/java/org/broadinstitute/hellbender/utils/test/FuncotatorTestUtils.java (hunk 2 above)
Comment (LeeTL1220) [reply src/test/java/org/broadinstitute/hellbender/utils/test/FuncotatorTestUtils.java]: Done.

File: src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (jonn-smith) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java]: Sounds good.  If anything, it's more flexible to make the VCF renderer behave more like the MAF renderer.

I'd need to review it to see how feasible that is.

I've added issue #5240 to track this.

[the remark this answers is not in the document]
Comment (LeeTL1220) [reply src/main/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorUtils.java]: Glad you filed the issue.  I would prefer the VCF renderer behave like the MAF renderer.  Then we could probably share a lot more of the code.

**Pull request merged**